File: selection sort/java/src/SelectionSort.java
public class SelectionSort {
    private int[] arr;

    SelectionSort() {
    }

    SelectionSort(int[] arr) {
        this.arr = arr;
    }

    void sort() {
        for (int i = 0; i < arr.length; i++) {
            int min = i;

            for (int j = i; j < arr.length; j++) {
                if (arr[min] > arr[j]) {
                    min = j;
                }
            }
            swap(arr, i, min);
        }
    }

    void sort(int[] arr) {
        this.arr = arr;
        sort();
    }

    private void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    void display() {
        if (arr != null) {

            System.out.print("[");
            for (int i = 0; i < arr.length; i++) {
                System.out.print(arr[i] + ((i < arr.length - 1) ? ", " : ""));
            }
            System.out.println("]");
        }
    }
}

File: selection sort/dart/selection_sort.dart
import 'dart:io';

class SelectionSort {
  List? _arr;
  SelectionSort({List? arr}) {
    if (arr != null) _arr = arr;
  }
  void sort({List? arr}) {
    if (arr != null) _arr = arr;
    for (int i = 0; i < _arr!.length; i++) {
      int min = i;
      for (int j = i; j < _arr!.length; j++) {
        if (_arr![min] > _arr![j]) {
          min = j;
        }
      }
      swap(i, min);
    }
  }

  void swap(int i, int min) {
    int temp = _arr![i];
    _arr![i] = _arr![min];
    _arr![min] = temp;
  }

  void display() {
    if (_arr != null) {
      stdout.write("[");
      for (int i = 0; i < _arr!.length; i++) {
        stdout.write('${_arr![i]}${i < _arr!.length - 1 ? ", " : ""}');
      }
      print("]");
    }
  }
}
